File: src (cpp)/animal/pisces/clownfish.cpp
#include "clownfish.h"

Clownfish::Clownfish(int _x, int _y, int _weight) : def_ID(14),
                                                    def_ratio_meat(30),
                                                    def_ratio_plant(40) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = false;
  is_water_animal = true;
  is_air_animal = false;
  weight = _weight;
}
void Clownfish::Interact() {
  std::cout << "The clownfish is not as funny as its name..." << std::endl;
}

File: src/virtualzoo/animal/pisces/Clownfish.java
package virtualzoo.animal.pisces;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Omnivora;

/**
 * Kelas Clownfish mendefinisikan atribut untuk clownfish.
 *
 * @author Holy Lovenia - 13515113
 * @version 3.0
 * @since 3.0
 */
public class Clownfish extends Pisces {

  /**
   * Constructor
   *
   * Menciptakan clownfish dengan posisi (locX,locY) dan berat animalWeight.
   *
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat clownfish.
   */
  public Clownfish(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Omnivora(animalWeight, 30, 40);
    ID = 14;
    position.setLocation(locX, locY);
    isLandAnimal = false;
    isWaterAnimal = true;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan clownfish.
   */
  public void interact() {
    System.out.println("The clownfish is not as funny as its name...");
  }
}
